src/Workspaces/Core/Portable/EmbeddedLanguages/LanguageServices/FallbackEmbeddedLanguage.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+using Microsoft.CodeAnalysis.EmbeddedLanguages.VirtualChars;
+using Microsoft.CodeAnalysis.LanguageServices;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages.LanguageServices
+{
+    /// <summary>
+    /// A 'fallback' embedded language that can classify normal escape sequences in 
+    /// C# or VB strings if no other embedded languages produce results.
+    /// </summary>
+    internal class FallbackEmbeddedLanguage : IEmbeddedLanguage
+    {
+        public int StringLiteralTokenKind { get; }
```

Comment:
>public [](start = 8, length = 6)

Do you expect these properties being  used for other aspects of embedded languages?
If not, I'd be tempted to push all these into the classifier.

We might even make the fallback classifier an inner type of this fallback language.

Reply:
yes.  That would be the expectation.  Because all the languages need to be able to quickly check if they're on a language token they can support lighting up on.

Reply:
Ok :-)
Making the classifier an inner type (and reducing accessibility on those properties) may still make sense. Up to you.

---
In reply to: [208743983](https://github.com/dotnet/roslyn/pull/28927#discussion_r208743983) [](ancestors = 208743983)

Reply:
done.